src/cpp/fastdds/rpc/ReplierImpl.cpp

```diff
@@ -137,6 +137,13 @@ ReturnCode_t ReplierImpl::take_request(
         return RETCODE_PRECONDITION_NOT_MET;
     }
 
+    // For taking the request is enough to check if the replier is fully matched
+    if (!is_fully_matched())
+    {
+        EPROSIMA_LOG_ERROR(REPLIER, "Trying to take a request with an unmatched requester");
```

Comment:
I think I should make this at least a warning since it could spuriously fail on applications shutdown